modules/plugins/statusline/feline/feline.nix

```diff
@@ -0,0 +1,144 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  inherit (builtins) elem;
+  inherit (lib.options) mkOption mkEnableOption literalExpression;
+  inherit (lib.types) str listOf attrs attrsOf anything either submodule;
+  inherit (lib.lists) optional;
+  inherit (lib.generators) mkLuaInline;
+  inherit (lib.nvim.types) mkPluginSetupOption;
+
+  conditionalRenderers = {
+    options = {
+      filetypes = mkOption {
+        type = listOf str;
+        default = [
+          "^NvimTree$"
+          "^neo-tree$"
+          "^startify$"
+          "^fugitive$"
+          "^fugitiveblame$"
+          "^qf$"
+          "^help$"
+        ];
+
+        description = "Filetypes in which to force render inactive statusline";
+      };
+
+      buftypes = mkOption {
+        type = listOf str;
+        default = ["^terminal$"];
+        description = "Buffer types in which to force render inactive statusline";
+      };
+
+      bufnames = mkOption {
+        type = listOf str;
+        default = [];
+        description = "Buffer names in which to force render inactive statusline";
+      };
+    };
+  };
+in {
+  options.vim.statusline.feline-nvim = {
+    enable = mkEnableOption "minimal, stylish and customizable statusline, statuscolumn, and winbar [feline.nvim]";
+    setupOpts = mkPluginSetupOption "feline-nvim" {
+      custom_providers = mkOption {
+        type = attrsOf anything;
+        default = {};
+        example = literalExpression ''
+          {
+            window_number = mkLuaInline ''''
+              function()
+                return tostring(vim.api.nvim_win_get_number(0))
+              end
+            '''';
+          }
+        '';
+
+        description = "User-defined feline provider functions";
+      };
+
+      theme = mkOption {
+        type = either str (attrsOf str);
+        default = {};
+        example = {
+          fg = "#fff";
+          bg = "#111";
+        };
+        description = ''
+          Either a string containing the color theme name or an attribute set of
+          strings, containing the colors.
+
+          The theme’s `fg` and `bg` values also represent the default foreground
+          and background colors, respectively.
+        '';
+      };
+
+      separators = mkOption {
+        type = listOf str;
```

Comment:
Aaand wrong again, this should be an attribute set, ref https://github.com/freddiehaddad/feline.nvim/blob/9f1313f61a75ec5ebe805fedd46bdc130c420963/lua/feline/defaults.lua#L17-L38